insights.go

```diff
@@ -16,10 +16,14 @@ func unregisterInsights() error {
 	return cmd.Run()
 }
 
-func insightsIsRegistered() bool {
+func insightsIsRegistered() (bool, error) {
 	cmd := exec.Command("/usr/bin/insights-client", "--status")
 
-	_ = cmd.Run()
+	err := cmd.Run()
 
-	return cmd.ProcessState.Success()
+	if err == nil {
```

Comment:
`ProcessState.Success()` returns true or false depending on the exit code of the command run. Couldn't we just return `cmd.ProcessState.Success()` and `err` here without needing to check if `err` is nil?

```go
err := cmd.Run()
return cmd.ProcessState.Success(), err
```

Then callers checking for whether err is nil or not will be able to act on the error or exit success state.

Reply:
Yeah, you are right. We can simplify the code a little bit.

Reply:
Why would calling `Success` cause a crash? I looked through the code for the method, and it appears to ultimately end up checking whether the return value of this method is zero or non-zero.

```go
func (w WaitStatus) ExitStatus() int {
	if !w.Exited() {
		return -1
	}
	return int(w>>shift) & 0xFF
}
```

Reply:
Ahhh yes, I see. In that case it makes sense that we need to check the error returned from `Run`. If it's an ExitError, that we know that it ran (unsuccessfully). If it's another kind of error, then we shouldn't access the `ProcessState` struct field, since it's likely nil.

Reply:
Good point, when `ExitError` is returned, then we should only report that system is not connected to Red Hat Insights. PR is updated.